cpu/lm4f120/periph/timer.c

```diff
@@ -35,132 +37,411 @@
  * @{
  */
 typedef struct {
-    void (*cb)(int);            /**< timeout callback */
+    void (*cb)(int);        /**< timeout callback */
+    unsigned int diviser;    /**< software clock diviser */
 } timer_conf_t;
 
 static timer_conf_t config[TIMER_NUMOF];
 /**@}*/
 
+#include "hw_timer.h"
+
+/* Missing from driverlib */
+static inline unsigned long
+PRIV_TimerPrescaleSnapshotGet(unsigned long ulBase, unsigned long ulTimer) {
+    return((ulTimer == TIMER_A) ? HWREG(ulBase + TIMER_O_TAPS) :
+           HWREG(ulBase + TIMER_O_TBPS));
+}
+
+static inline unsigned long long _scaled_to_ll_value(unsigned int uncorrected, unsigned int diviser)
+{
+    const unsigned long long scaledv = (unsigned long long) uncorrected * diviser;
+    return scaledv;
+}
+
+static inline unsigned int _llvalue_to_scaled_value(unsigned long long corrected, unsigned int diviser)
+{
+    const unsigned long long scaledv = corrected / diviser;
+    return scaledv;
+}
+
 int timer_init(tim_t dev, unsigned int us_per_tick, void (*callback)(int))
 {
-    if (dev == TIMER_0) {
-        config[dev].cb = callback;                          /* User Function */
-        ROM_SysCtlPeripheralEnable(SYSCTL_PERIPH_WTIMER0);  /* Activate Timer0 */
-        WTIMER0_CTL_R &= ~0x00000001;                       /* Disable timer0A during setup */
-        WTIMER0_CFG_R  = TIMER_CFG_16_BIT;
-        WTIMER0_TAMR_R = TIMER_TAMR_TAMR_PERIOD;            /* Configure for periodic mode */
-        WTIMER0_TAPR_R = TIMER_0_PRESCALER;                 /* 1us timer0A */
-        WTIMER0_ICR_R  = 0x00000001;                        /* clear timer0A timeout flag */
-        WTIMER0_IMR_R |= 0x00000001;                        /* arm timeout interrupt */
-        ROM_IntPrioritySet(INT_WTIMER0A, 32);
-        timer_irq_enable(dev);
-        timer_start(dev);
-        DEBUG("startTimeout Value=0x%lx\n", ROM_TimerValueGet(WTIMER0_BASE, TIMER_A));
-        return 1;
-    }
-    return -1;
+    if (dev >= TIMER_NUMOF){
+        return -1;
+    }
+
+    config[dev].cb = callback;                          /* User Function */
+    config[dev].diviser = us_per_tick * ROM_SysCtlClockGet()/1000000;
+
+    unsigned int sysctl_timer;
```

Comment:
I think you will be required to declare variables at the start of the scope block. Though this is permitted in C99 but I was asked to adhere to traditional C style during one of the review. 
But anyway, I will let PR reviewers @thomaseichinger  @kaspar030 @PeterKietzmann comment on this.